### PR TITLE
Handle invalid cancellation signals in trymerge

### PR DESCRIPTION
**Pull request by huydhn**
This change is needed after https://github.com/pytorch/test-infra/pull/4579 and https://github.com/pytorch/test-infra/pull/4610.  All invalid cancelled signals have been removed from Dr.CI and HUD.  So trymerge should ignore them accordingly for a consistent experience.

### Testing

https://github.com/pytorch/pytorch/pull/110367#issuecomment-1750099960 is the PR where a bunch of invalid cancelled signals showed up and blocked merges

### CODE REVIEW

**Review by clee2000 — APPROVED** [review]
I was always under the impression that the graphql query only returned the most recent workflow but ig not.
I also have some miild concerns about jobs that don't every go to rockset like the cla check or the meta internal changes check, but I've also never seen them get cancelled.

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -1641,6 +1661,8 @@ def insert(
         rockset_results = get_rockset_results(head_sha, merge_base)
         for rockset_result in rockset_results:
             name = f"{rockset_result['workflow_name']} / {rockset_result['name']}"
+            rockset_result["full_name"] = name
```
[the remark this answers is not in the document]
Comment (huydhn) [reply .github/scripts/trymerge.py:1664]: You're right.  I just haven’t refactored the code to do that yet, so this section is still here, but they will go away soon.  I plan that as the last PR in the list here https://github.com/pytorch/pytorch/pull/110133 to make broken trunk results from Dr.CI the source of truth in trymerge.

**Comment by huydhn** [comment]
> I was always under the impression that the graphql query only returned the most recent workflow but ig not. I also have some miild concerns about jobs that don't every go to rockset like the cla check or the meta internal changes check, but I've also never seen them get cancelled.

I'll add a simple check for that, these jobs are not from Rockset and they don't have a full `workflow / job` name, and that can be used in the check.

**Review by ZainRizvi — APPROVED** [review]

**Pull request closed without merging**